nano/node/wallet.cpp

```diff
@@ -1579,6 +1602,38 @@ void nano::wallets::clear_send_ids (nano::transaction const & transaction_a)
 	assert (status == 0);
 }
 
+void nano::wallets::compute_reps ()
+{
+	std::lock_guard<std::mutex> lock (mutex);
+	auto ledger_transaction (node.store.tx_begin_read ());
+	auto transaction (tx_begin_read ());
+	for (auto i (items.begin ()), n (items.end ()); i != n; ++i)
+	{
+		auto & wallet (*i->second);
+		decltype (wallet.representatives) representatives_l;
+		for (auto ii (wallet.store.begin (transaction)), nn (wallet.store.end ()); ii != nn; ++ii)
+		{
+			auto account (ii->first);
+			if (node.ledger.weight (ledger_transaction, account) >= node.config.vote_minimum.number ())
+			{
+				representatives_l.insert (account);
+			}
+		}
+		std::lock_guard<std::mutex> representatives_lock (wallet.representatives_mutex);
+		wallet.representatives.swap (representatives_l);
+	}
+}
+
+void nano::wallets::ongoing_compute_reps ()
+{
+	compute_reps ();
+	auto & node_l (node);
+	auto compute_delay (nano::nano_network == nano::nano_networks::nano_test_network ? std::chrono::milliseconds (10) : std::chrono::milliseconds (15 * 60 * 1000)); // Representation drifts quickly on the test network but very slowly on the live network
```

Comment:
I wonder if `compute_reps` on a node with many wallets and a ton of accounts warrants a less frequent scan than 15 seconds (or a config option) - or is the scan fast enough?

Reply:
15 minutes

Reply:
thanks 😆